Mathlib/AlgebraicGeometry/EllipticCurve/DivisionPolynomial/Basic.lean

```diff
@@ -322,25 +342,42 @@ lemma Ψ_three : W.Ψ 3 = C W.Ψ₃ := by
 lemma Ψ_four : W.Ψ 4 = C W.preΨ₄ * W.ψ₂ := by
   erw [Ψ_ofNat, preΨ'_four, if_pos <| by decide]
 
-lemma Ψ_odd (m : ℕ) : W.Ψ (2 * (m + 2) + 1) =
-    W.Ψ (m + 4) * W.Ψ (m + 2) ^ 3 - W.Ψ (m + 1) * W.Ψ (m + 3) ^ 3 +
-      W.toAffine.polynomial * (16 * W.toAffine.polynomial - 8 * W.ψ₂ ^ 2) * C
-        (if Even m then W.preΨ' (m + 4) * W.preΨ' (m + 2) ^ 3
-          else -W.preΨ' (m + 1) * W.preΨ' (m + 3) ^ 3) := by
-  repeat erw [Ψ_ofNat]
-  simp_rw [preΨ'_odd, if_neg (m + 2).not_even_two_mul_add_one, Nat.even_add_one, ite_not]
-  split_ifs <;> C_simp <;> rw [C_Ψ₂Sq] <;> ring1
-
-lemma Ψ_even (m : ℕ) : W.Ψ (2 * (m + 3)) * W.ψ₂ =
+lemma Ψ_even' (m : ℕ) : W.Ψ (2 * (m + 3)) * W.ψ₂ =
     W.Ψ (m + 2) ^ 2 * W.Ψ (m + 3) * W.Ψ (m + 5) - W.Ψ (m + 1) * W.Ψ (m + 3) * W.Ψ (m + 4) ^ 2 := by
   repeat erw [Ψ_ofNat]
```

Comment:
Why is `erw` needed? Is there perhaps a bit of API missing, making `rw` work smoothly?

Reply:
Some of these are just for the sake of my sanity, e.g. converting 4 - 1 to 3 by `rw [show (4 - 1 : ℤ) = 3 by rfl]` etc.

I can add these to the bottom of [this file](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Nat/Cast/Defs.html) though, what do you think?

Reply:
Wait, I have to find a different home for those lemmas, since I don't think file imports `AddGroup`...